File: server/models/trilium.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class TriliumStatus(BaseModel):
    online: bool
    appVersion: Optional[str] = None
    dbVersion: Optional[int] = None
    syncVersion: Optional[int] = None
    buildDate: Optional[datetime] = None
    buildRevision: Optional[str] = None
    dataDirectory: Optional[str] = None
    clipperProtocolVersion: Optional[str] = None
    utcDateTime: Optional[datetime] = None

File: tr-util/src/types/trilium.ts
export type TriliumStatus = ({
    online: true;
    appVersion: string;
    dbVersion: number;
    syncVersion: number;
    buildDate: string;
    buildRevision: string;
    dataDirectory: string;
    clipperProtocolVersion: string;
    utcDateTime: string;
} | {
    online: false
}) & {
    url: string;
}